pos_order_to_sale_order/static/src/js/pos_order_to_sale_order.js

```diff
@@ -0,0 +1,173 @@
+/******************************************************************************
+    Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+    @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+ *****************************************************************************/
+
+openerp.pos_order_to_sale_order = function(instance, local) {
+
+    "use strict";
+
+    var module = instance.point_of_sale;
+    var _t = instance.web._t;
+
+    /*************************************************************************
+        New Widget CreateSaleOrderButtonWidget:
+            * On click, check if there is a customer defined,
+              ask confirmation call server to create sale order, and delete
+              the current order. 
+    */
+    module.CreateSaleOrderButtonWidget = module.PosBaseWidget.extend({
+        template: 'CreateSaleOrderButtonWidget',
+
+        init: function(parent, options){
+            console.log(options);
+            this._super(parent, options);
+            this.sale_order_state = options.sale_order_state;
+            if (this.sale_order_state == 'draft') {
+                this.display_text = _t("Create Draft Order");
+                this.confirmation_message = _t('Create Draft Sale Order and discard the current PoS Order ?');
+                this.confirmation_comment = _t("This operation will permanently discard the current PoS Order and create a draft Sale Order, based on the current order lines. Note if you had manually changed unit prices for some products, this changes will not been taken into account in the sale order.");
```

Comment:
had changed->have changed
(shouldnot this limitation be highlighted in the README BTW)

Reply:
It is writtent in the readme.